src/utils/locks.js

```diff
@@ -14,18 +14,20 @@ export async function lock(packages: Array<Package>) {
   logger.info('Attempting to get locks for all packages');
 
   for (let pkg of packages) {
-    let {name, version} = pkg.config;
+    let { name, version } = pkg.config;
     let promise = npm.infoAllow404(name).then(response => {
-      if (response.response !== '404') {
-        const pkgInfo = response.data || {};
+      if (response.published) {
```

Comment:
Will need tests for the rest of these util functions **soon** and this wouldnt have happened :(

Would adding an explicit return type to infoAllow404 have helped as well?